internal/pkg/handlers/containerService.go

```diff
@@ -0,0 +1,332 @@
+			ID:                  currentItem.ID + "/" + display,
+			Namespace:           "swagger",
+			Name:                display,
+			Display:             display,
+			ExpandURL:           resourceType.Endpoint.TemplateURL, // all fixed template URLs
+			ItemType:            SubResourceType,
+			SwaggerResourceType: &resourceType,
+			Metadata: map[string]string{
+				"SwaggerAPISetID": currentItem.ID,
+			},
+		})
+	}
+
+	return ExpanderResult{
+		Err:               nil,
+		Response:          "TODO - what should go here?",
+		SourceDescription: "AzureKubernetesServiceExpander request",
+		Nodes:             newItems,
+		IsPrimaryResponse: true,
+	}
+}
+
+func (e *AzureKubernetesServiceExpander) createAPISetForCluster(ctx context.Context, clusterID string) (*SwaggerAPISetContainerService, error) {
+	kubeConfig, err := e.getClusterConfig(ctx, clusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	// NOTE - at the time of writing the AKS API returns a single cluster/user
+	// so we're not fully parsing the config, just taking the first user and cluster
+
+	httpClient, err := e.getHTTPClientFromConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	serverURL := kubeConfig.Clusters[0].Cluster.Server
+
+	swaggerResourceTypes, err := e.getSwaggerResourceTypes(*httpClient, serverURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// Register the swagger config so that the swagger expander can take over
+	apiSet := NewSwaggerAPISetContainerService(swaggerResourceTypes, *httpClient, clusterID+"/<k8sapi>", serverURL)
+	return &apiSet, nil
+}
+func (e *AzureKubernetesServiceExpander) getAPISetForCluster(clusterID string) *SwaggerAPISetContainerService {
+
+	swaggerAPISet := GetSwaggerResourceExpander().GetAPISet(clusterID + "/<k8sapi>")
+	if swaggerAPISet == nil {
+		return nil
+	}
+	apiSet := (*swaggerAPISet).(SwaggerAPISetContainerService)
+	return &apiSet
+}
+
+func (e *AzureKubernetesServiceExpander) getSwaggerResourceTypes(httpClient http.Client, serverURL string) ([]swagger.ResourceType, error) {
+
+	var swaggerResourceTypes []swagger.ResourceType
+
+	url := serverURL + "/openapi/v2"
+
+	response, err := httpClient.Get(url)
+	if err != nil {
+		return swaggerResourceTypes, err
+	}
+
+	defer response.Body.Close() //nolint: errcheck
+	buf, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return swaggerResourceTypes, err
+	}
+
+	spec := json.RawMessage(buf)
+	doc, err := loads.Analyzed(spec, "")
+	if err != nil {
+		return swaggerResourceTypes, err
+	}
+
+	config := swagger.Config{
+		AdditionalGetPaths: []string{
+			"/api/v1/watch/", // add a placeholder for /api/v1/watch/* as otherwise they end up directly under /api/v1 and show as duplicates of /api/v1/nodes etc
+		},
+		SuppressAPIVersion: true,
+	}
+	var paths []*swagger.Path
+	paths, err = swagger.MergeSwaggerDoc(paths, &config, doc, false)
+	if err != nil {
+		return swaggerResourceTypes, err
+	}
+
+	// uncomment this code to help debug the generated API paths!
```

Comment:
@lawrencegripper what's the wider story about enabling debug features - do we drive all this behaviour via the `--debug` flag? Do we need to look for build macro alternatives or do you think uncomment and rebuild is sufficient?

Reply:
Just pushed an update that outputs this based on whether debug is set